developers/documentation/index.md

```diff
@@ -0,0 +1,78 @@
+
+## Document types
+napari's documentation is based on the [Diátaxis Framework](https://diataxis.fr/) and consists of four main types of documents, each with their own section on the website.
+
+![Diátaxis framework placing tutorials, how-to guides, explanations and references on a two axis diagram](https://diataxis.fr/_images/diataxis.png)
+
+**Tutorials** are a contained lesson designed to help the user achieve understanding of a concept, feature, workflow or process through practice. They should be composed of a set of concrete, sequential steps that guide the user towards a clearly defined result. Typically, tutorials are sufficiently detailed to provide a basis for further learning without getting too tangled in explanations, and are aimed at users who don't necessarily know what they want to do and what's available to them.
+
+**How-tos** are also concrete steps that help the user to achieve a defined result. Unlike tutorials, users following a how-to already know *what* they want to achieve, they're just not sure how to do it. How-tos should list steps with only the minimal detail required to repeat them.
+
+**Explanations** focus on theoretical and concept understanding. They are inevitably more in depth than both tutorials and how-tos, and open up the floor for discussion.
+
+### Picking your document
+
+New documentation might be useful when we develop a new viewer or layer interaction, add an entirely new feature to napari's API, change the way things are built under the hood or use napari for a novel application.
+
+Where possible, small new features or interactions with layers or the viewer itself should be added to the existing tutorials and how-tos to make sure they are discoverable. 
+For example, auto-scaling contrast limits on a layer should be documented in the Viewer tutorial. 
+If a feature is fairly complex it might warrant its own document, e.g. hooking up your own callbacks to layer events or performance monitoring your Python scripts.
+
+To pick your document, you should consider how much detail you want to communicate and to whom. 
+For example "I want to document how this feature has been built for future developers, so that it's easy to maintain" - this sounds like you're wanting to communicate complex, in-depth information for developers, and would require an explanation style document. 
+"I want users to know how to toggle the theme" is a set of very quick steps for users who know what they want - to change the theme, so this should be a how-to. 
+"I want to give biologists an overview of how to perform segmentation in napari" is a broader lesson that may require some explanations for users and pointing out a variety of features, so a tutorial would be suitable. 
+Of course, you might want to write more than one document depending on who you're targetting and what information you think is relevant.
+
+
+## Documentation templates
```

Comment:
This section needs links that say "Click here to download the template!" (that you know, download the template when you click on them). I see links below, but they look like they'll just open up another browser tab instead of downloading the template.